src/CardList/index.js

```diff
@@ -117,23 +122,71 @@ export default class ImageList extends Component {
     return <Text style={headerStyles}>{listHeader.header}</Text>
   }
 
+  filterItems(items) {
+    return items.filter((itm) => {
+      if (!this.state.currentQuery) {
+        return true
+      }
+      if (
+        itm.title.text &&
+        itm.title.text.indexOf(this.state.currentQuery) >= 0
+      ) {
+        return true
+      } else if (
+        itm.subtitle.text &&
+        itm.subtitle.text.indexOf(this.state.currentQuery) >= 0
+      ) {
+        return true
+      } else if (
+        itm.body.text &&
+        itm.body.text.indexOf(this.state.currentQuery) >= 0
+      ) {
+        return true
+      }
+    })
+  }
+
   render() {
-    let { cardLayout } = this.props
+    let { cardLayout, searchBar, items } = this.props
+    const { currentQuery } = this.state
     let wrap = [styles.wrap]
 
+    const newItems = this.filterItems(items)
+
+    const notFound = newItems.length === 0
+
     if (cardLayout === 'grid') {
       return (
-        <View style={wrap}>
-          {this.renderHeader()}
-          {this.renderGrid()}
-        </View>
+        <>
+          <View style={wrap}>
+            {this.renderHeader()}
+            <SearchBarWrapper
+              searchBar={this.props.searchBar}
```

Comment:
It just occurred to me - don't you already get the `notFoundText` inside here? That way you don't have to pass in an extra prop.